### PR TITLE
Add Super-Linter as GitHub Action

### PR DESCRIPTION
**Pull request by razvand**
Use Super-Linter to validate code base. It will validate both code files (in different programming languages, quite a lot are supported) and Markdown files.

Markdown validation is configured in `config/config.json` and `rules/rules.js`.

Validation with Super-Linter is triggered for push actions and for pull requests.

### CODE REVIEW

File: .github/workflows/rules/md101.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    new WordPattern("jps"),
+    new WordPattern("name=value"),
+    new WordPattern("key=value"),
+    new WordPattern("time:value"),
+    new WordPattern("atsd.log"),
+    new WordPattern("start.log"),
+    new WordPattern("logback.xml"),
+    new WordPattern("graphite.conf"),
+    new WordPattern("command_malformed.log"),
+    new WordPattern("stdout"),
```
Comment (teodutu) [inline .github/workflows/rules/md101.js:25]:
```suggestion
    new WordPattern("stdin"),
    new WordPattern("stdout"),
```

File: .github/workflows/rules/md102.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    "POST",
+    "PUT",
+    "PATCH",
+    "DELETE",
+    "Content-Type",
+    "Content-Encoding",
+    "User-Agent",
+    "200 OK",
+    "401 Unauthorized",
+    "403 Forbidden",
```
Comment (teodutu) [inline .github/workflows/rules/md102.js:12]:
```suggestion
    "403 Forbidden",
    "404 Not Found",
```

File: .github/workflows/rules/md101.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    new WordPattern("stderr"),
+    new WordPattern("SIGTERM"),
```
Comment (teodutu) [inline .github/workflows/rules/md101.js:27]:
```suggestion
    new WordPattern("SIGTERM"),
    new WordPattern("SIGINT"),
    new WordPattern("SIGSEGV"),
    new WordPattern("SIGKILL"),
    new WordPattern("SIGABRT"),
    new WordPattern("SIGILL"),
    new WordPattern("SIGCHLD"),
```
And

File: .github/workflows/rules/md101.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    new WordPattern(".xml", { noWordBoundary: true }),
+    new WordPattern(".jar", { noWordBoundary: true }),
+    new WordPattern(".gz", { noWordBoundary: true }),
+    new WordPattern(".tar.gz", { noWordBoundary: true }),
+    new WordPattern(".zip", { noWordBoundary: true }),
+    new WordPattern(".txt", { noWordBoundary: true }),
+    new WordPattern(".csv", { noWordBoundary: true }),
+    new WordPattern(".json", { noWordBoundary: true }),
+    new WordPattern(".pdf", { noWordBoundary: true }),
+    new WordPattern(".html", { noWordBoundary: true })
```
Comment (teodutu) [inline .github/workflows/rules/md101.js:39]:
```suggestion
    new WordPattern(".html", { noWordBoundary: true }),
    new WordPattern(".c", { noWordBoundary: true }),
    new WordPattern(".cpp", { noWordBoundary: true }),
    new WordPattern(".py", { noWordBoundary: true }),
    new WordPattern(".d", { noWordBoundary: true }),
    new WordPattern(".java", { noWordBoundary: true }),
    new WordPattern(".go", { noWordBoundary: true }),
    new WordPattern(".o", { noWordBoundary: true }),
    new WordPattern(".a", { noWordBoundary: true }),
    new WordPattern(".so", { noWordBoundary: true })
```
And any other language or file format we use.

**Review by teodutu — CHANGES_REQUESTED** [review]
If my understanding of the `md*.js` files is correct, then I suggested some useful additions to them.

File: .github/workflows/rules/md101.js [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    new WordPattern(".c", { noWordBoundary: true })
+    new WordPattern(".cpp", { noWordBoundary: true })
+    new WordPattern(".h", { noWordBoundary: true })
+    new WordPattern(".hpp", { noWordBoundary: true })
+    new WordPattern(".py", { noWordBoundary: true })
```
Comment (teodutu) [inline .github/workflows/rules/md101.js:75]: Why not `.d` as well? Can it trigger false positives?

**Review by teodutu — APPROVED** [review]
Looks fine. Those lists of constructs that need to be wrapped in backticks can be enhanced in the future when the need arises. But mind my comment about `.d` files.

File: .github/workflows/rules/md101.js (hunk 5 above)
Comment (razvand) [reply .github/workflows/rules/md101.js:80]: @teodutu , added.

**Review by Sergiu121 — APPROVED** [review]

**Comment by teodutu** [comment]
Running the linter [1] triggers this error: `Unable to resolve action `github/super-linter@v4`, unable to find version `v4`. Maybe try a lower version?

[1] https://github.com/open-education-hub/operating-systems-oer/actions/runs/3151469773/jobs/5125453900

**Pull request merged**